cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java

```diff
@@ -611,4 +632,20 @@ public boolean meetCompletenessRequirements(List<String> goalNames) {
     }
     return goalsByPriority;
   }
+
+  /**
+   * Check whether all hard goals are included in provided goal list.
+   *
+   * @param goals A list of goals.
+   * @param skipHardGoalCheck True if hard goal checking is not needed.
+   */
+  private void goalListSanityCheck(List<String> goals, boolean skipHardGoalCheck) {
+    // If goals is null, skip hard goal check since all default goals will be picked to calculate proposal.
```

Comment:
If default goals config is null, then all goals will be picked up.
Maybe we can remove this comment to make this function self-contained -- i.e. does not depend on how another function works. We can update the javadoc to mention what this function does if `goals` is null.